tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -41,83 +41,93 @@
 
 
 class OutputFormat(object):
-  """An enum used to list the valid output formats for API calls."""
-  JSON = 'json'
-  CSV = 'csv'
+    """An enum used to list the valid output formats for API calls."""
+
+    JSON = "json"
+    CSV = "csv"
 
 
 class ScalarsPlugin(base_plugin.TBPlugin):
-  """Scalars Plugin for TensorBoard."""
-
-  plugin_name = metadata.PLUGIN_NAME
-
-  def __init__(self, context):
-    """Instantiates ScalarsPlugin via TensorBoard core.
-
-    Args:
-      context: A base_plugin.TBContext instance.
-    """
-    self._multiplexer = context.multiplexer
-    self._db_connection_provider = context.db_connection_provider
-    if context.flags and context.flags.generic_data == 'true':
-      self._data_provider = context.data_provider
-    else:
-      self._data_provider = None
-
-  def get_plugin_apps(self):
-    return {
-        '/scalars': self.scalars_route,
-        '/tags': self.tags_route,
-    }
-
-  def is_active(self):
-    """The scalars plugin is active iff any run has at least one scalar tag."""
-    if self._data_provider:
-      # We don't have an experiment ID, and modifying the backend core
-      # to provide one would break backward compatibility. Hack for now.
-      return True
-
-    if self._db_connection_provider:
-      # The plugin is active if one relevant tag can be found in the database.
-      db = self._db_connection_provider()
-      cursor = db.execute('''
+    """Scalars Plugin for TensorBoard."""
+
+    plugin_name = metadata.PLUGIN_NAME
+
+    def __init__(self, context):
+        """Instantiates ScalarsPlugin via TensorBoard core.
+
+        Args:
+          context: A base_plugin.TBContext instance.
+        """
+        self._multiplexer = context.multiplexer
+        self._db_connection_provider = context.db_connection_provider
+        if context.flags and context.flags.generic_data == "true":
+            self._data_provider = context.data_provider
+        else:
+            self._data_provider = None
+
+    def get_plugin_apps(self):
+        return {
+            "/scalars": self.scalars_route,
+            "/tags": self.tags_route,
+        }
+
+    def is_active(self):
+        """The scalars plugin is active iff any run has at least one scalar
+        tag."""
+        if self._data_provider:
+            # We don't have an experiment ID, and modifying the backend core
+            # to provide one would break backward compatibility. Hack for now.
+            return True
+
+        if self._db_connection_provider:
+            # The plugin is active if one relevant tag can be found in the database.
+            db = self._db_connection_provider()
+            cursor = db.execute(
+                """
         SELECT
```

Comment:
SGTM. I’d prefer to do this in a change separate to the autogenerated
reformatting. I can do all the DB cursor ones by hand, and maybe wrangle
up a tiny Awk script to find others.

Reply:
Okay, here’s a script to find offenders (needs Python 3.8 for recent
changes to the `ast` module):

```python
import ast
import glob
import os


def main():
    for filename in glob.glob("tensorboard/**/*.py", recursive=True):
        with open(filename) as infile:
            source = infile.read()
        sourcelines = source.splitlines()
        module = ast.parse(source, filename=filename)
        strings = [x for x in ast.walk(module) if isinstance(x, ast.Str)]
        for node in strings:
            base = _indent_of(sourcelines[node.lineno - 1])
            lines = node.s.splitlines()
            if len(lines) <= 1:
                continue
            if any(_indent_of(x) < base for x in lines[1:] if x.strip()):
                print("%s:%d" % (filename, node.lineno))


def _indent_of(line):
    n = 0
    for c in line:
        if c != " ":
            break
        n += 1
    return n


if __name__ == "__main__":
    main()
```

We have 235 such string literals across lots of files. With a quick spot
check, looks like we have:

  - True positives:
      - SQL queries
      - pbtxt literals
      - a few docstrings that `docformatter` didn’t fix because they weren’t well-formatted in the first place
  - False positives:
      - Flags in `core_plugin.py`, which have always been at column 0
      - Some strings with `\n` sequences in them, which `ast` doesn’t
        distinguish from multi-line strings with newlines

I can probably just take a pass through all of these and fix them. There
aren’t that many, but they’ll require manual inspection to make sure
that the change is sound.

Reply:
Filed #3034.